File: frontend/src/types/triage.ts
import type { TicketPriority, TicketStatus } from "./ticket";

export type TriageSummary = {
  total_open: number;
  high_priority: number;
  low_confidence: number;
  possible_incidents: number;
};

export type TriageTicket = {
  id: string;
  title: string;
  description: string;
  status: TicketStatus;
  assigned_department: string | null;
  category: string | null;
  category_label: string | null;
  priority: TicketPriority | null;
  priority_score: number | null;
  category_confidence: number | null;
  suggested_department: string | null;
  created_at: string;
  updated_at: string;
};

export type IncidentRelatedTicket = {
  ticket_id: string;
  title: string;
  similarity: number;
  reason: string;
};

export type IncidentGroupSuggestion = {
  group_key: string;
  title: string;
  category: string | null;
  average_similarity: number;
  related_count: number;
  related_tickets: IncidentRelatedTicket[];
  recommendation: string;
};

export type RoutingRecommendation = {
  ticket: TriageTicket;
  recommended_department: string;
  reason: string;
};

export type TriageOverview = {
  summary: TriageSummary;
  critical_queue: TriageTicket[];
  low_confidence_cases: TriageTicket[];
  possible_incident_groups: IncidentGroupSuggestion[];
  routing_recommendations: RoutingRecommendation[];
  recent_tickets: TriageTicket[];
};

File: backend/app/schemas/triage.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.ticket import TicketStatus


class TriageSummary(BaseModel):
    total_open: int
    high_priority: int
    low_confidence: int
    possible_incidents: int


class TriageTicketItem(BaseModel):
    id: UUID
    title: str
    description: str
    status: TicketStatus
    assigned_department: str | None = None
    category: str | None = None
    category_label: str | None = None
    priority: str | None = None
    priority_score: int | None = None
    category_confidence: float | None = None
    suggested_department: str | None = None
    created_at: datetime
    updated_at: datetime


class IncidentRelatedTicket(BaseModel):
    ticket_id: str
    title: str
    similarity: float
    reason: str


class IncidentGroupSuggestion(BaseModel):
    group_key: str
    title: str
    category: str | None = None
    average_similarity: float
    related_count: int
    related_tickets: list[IncidentRelatedTicket]
    recommendation: str


class RoutingRecommendation(BaseModel):
    ticket: TriageTicketItem
    recommended_department: str
    reason: str


class TriageOverviewResponse(BaseModel):
    summary: TriageSummary
    critical_queue: list[TriageTicketItem]
    low_confidence_cases: list[TriageTicketItem]
    possible_incident_groups: list[IncidentGroupSuggestion]
    routing_recommendations: list[RoutingRecommendation]
    recent_tickets: list[TriageTicketItem]
